### PR TITLE
Rename SWIFT_ENABLE_COMPATIBILITY_OVERRIDES -> SWIFT_STDLIB_SUPPORT_BACK_DEPLOYMENT and avoid building more back-deployment stdlib parts when not set

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

File: stdlib/CMakeLists.txt [hunk 1]
```diff
@@ -89,8 +89,8 @@ option(SWIFT_ENABLE_MODULE_INTERFACES
        "Generate .swiftinterface files alongside .swiftmodule files"
        "${SWIFT_STDLIB_STABLE_ABI}")
 
-option(SWIFT_ENABLE_COMPATIBILITY_OVERRIDES
-       "Support back-deploying compatibility fixes for newer apps running on older runtimes."
+option(SWIFT_STDLIB_SUPPORT_BACK_DEPLOYMENT
+       "Support back-deployment of the stdlib to older OS versions."
```
Comment (mikeash) [inline stdlib/CMakeLists.txt:93]: This should say something like, "Support back-deployment of built binaries to older OS versions" or "to older stdlib versions" or something like that. We don't support back-deployment of the stdlib itself.

**Review by mikeash — APPROVED** [review]

**Review by compnerd — APPROVED** [review]

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by al45tair — APPROVED** [review]

**Comment by kubamracek** [comment]
@swift-ci please test macOS platform

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test Linux platform

**Comment by kubamracek** [comment]
@swift-ci please test Windows platform

**Comment by kubamracek** [comment]
Note to self: This seems to fail to build the freestanding/minimal preset now. Don't merge before fixing that.

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
preset=stdlib_S_standalone_minimal_macho_x86_64,build,test
@swift-ci please test with toolchain and preset

**Comment by kubamracek** [comment]
preset=stdlib_S_standalone_minimal_macho_x86_64,build,test
@swift-ci please test with toolchain and preset

**Comment by kubamracek** [comment]
@swift-ci please test macOS platform

**Comment by kubamracek** [comment]
preset=stdlib_S_standalone_minimal_macho_x86_64,build,test
@swift-ci please test with toolchain and preset

**Comment by kubamracek** [comment]
@swift-ci please test macOS platform

**Pull request merged**